### ISSUE TITLE
Logback - logback.xml, reading properties from application.yml fails ONLY in production

### ISSUE THREAD

**Issue by gsartori** [grails/grails-core#12874]
### Expected Behavior

Running the app a folder sould be created like the follow:

`appname/logs/appname.log`

The following configuration works when running the app in DEV on IntelliJ IDEA but fails when the app is run as .jar in PROD:

--
logback.xml
--

<?xml version="1.0" encoding="UTF-8"?>
<configuration>

    <conversionRule conversionWord="clr" converterClass="org.springframework.boot.logging.logback.ColorConverter" />
    <conversionRule conversionWord="wex" converterClass="org.springframework.boot.logging.logback.WhitespaceThrowableProxyConverter" />

    <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
        <encoder>
            <charset>UTF-8</charset>
            <pattern>%clr(%d{yyyy-MM-dd HH:mm:ss.SSS}){faint} %clr(%5p) %clr(---){faint} %clr([%15.15t]){faint} %clr(%-40.40logger{39}){cyan} %clr(:){faint} %m%n%wex</pattern>
        </encoder>
    </appender>

    <property resource="application.yml" />
    <appender name="FILE" class="ch.qos.logback.core.rolling.RollingFileAppender">
        <file>${info.app.name}/logs/${info.app.name}.log</file>

        <rollingPolicy class="ch.qos.logback.core.rolling.SizeAndTimeBasedRollingPolicy">
            <fileNamePattern>${info.app.name}/logs/archive/${info.app.name}.%d{yyyy-MM-dd}.%i.log.zip</fileNamePattern>
            <cleanHistoryOnStart>true</cleanHistoryOnStart>
            <maxHistory>10</maxHistory>
            <maxFileSize>10MB</maxFileSize>
            <totalSizeCap>100GB</totalSizeCap>
        </rollingPolicy>

        <encoder>
            <charset>UTF-8</charset>
            <pattern>%d{yyyy-MM-dd HH:mm:ss.SSS} %5p --- [%15.15t] %-40.40logger{39} : %m%n%wex</pattern>
        </encoder>
    </appender>

    <root level="ERROR">
        <appender-ref ref="STDOUT" />
        <appender-ref ref="FILE" />
    </root>

</configuration>

### Actual Behaviour

The following folder structure gets created in PROD (running the app as jar):

`info.app.name_IS_UNDEFINED/logs/info.app.name_IS_UNDEFINED.log`

### Steps To Reproduce

1. `grails create-app test`
2. copy the attached [logback.xml.zip](https://github.com/grails/grails-core/files/10731157/logback.xml.zip) into `test/grails-app/conf/`
3. `./gradlew bootJar`
4. `cd test/build/libs`
5. `java -jar test-0.1.jar`

### Environment Information

openjdk version "17.0.5" 2022-10-18
OpenJDK Runtime Environment GraalVM CE 22.3.0 (build 17.0.5+8-jvmci-22.3-b08)
OpenJDK 64-Bit Server VM GraalVM CE 22.3.0 (build 17.0.5+8-jvmci-22.3-b08, mixed mode, sharing)
Groovy Version: 4.0.8 JVM: 17.0.5 Vendor: GraalVM Community OS: Mac OS X
Grails Version: 5.2.5
JVM Version: 17.0.5

### Version

5.2.5

**Comment by gsartori** [comment]
Adding to the issue, it happens only to the properies expanded by grails, like the following:

```
info:
    app:
        name: '@info.app.name@'
        version: '@info.app.version@'
        grailsVersion: '@info.app.grailsVersion@'
```

it works in PROD when configured as:

`info.app.name: my-app-name`

**Comment by shinjiorisu** [comment]
This is still relevant in Grails 7.0.2

**Comment by gsartori** [comment]
This is still relevant in 7.1.0

**Comment by jdaugherty** [comment]
Root Cause (Two-part)                                                                                                                                                                                                                  
                                                                                                                                                                                                                                           
  1. Logback can't parse YAML natively.                                                                                                                                                                                                    
  The user's <property resource="application.yml" /> directive tells Logback to load a classpath resource as a Java .properties file. Logback has zero YAML support — it reads key=value format only. So ${info.app.name} resolves to
  info.app.name_IS_UNDEFINED in all environments, not just production. The fact that it seems to work in development is likely coincidence (e.g., the JAR name matches, or the user isn't looking closely at dev logs).                    
                                                                                                                                                                                                                                         
  2. Timing: logback initializes before Spring.                                                                                                                                                                                            
  Even if the YAML were parseable, Spring Boot initializes the logging subsystem before it loads application.yml into the Environment. There's no code in Grails (ExternalConfigRunListener, GrailsApplicationPostProcessor, etc.) that  
  feeds application properties into Logback at startup.                                                                                                                                                                                    
                                                                                                                                                                                                                                         
  The Correct Fix (for users)                                                                                                                                                                                                              
                                                                                                                                                                                                                                         
  Spring Boot's Logback extension provides <springProperty> specifically for this. In a logback-spring.xml (not logback.xml):                                                                                                              
                                                                                                                                                                                                                                         
                <springProperty scope="local" name="appName" source="info.app.name" defaultValue="myapp"/>                                                                                                                                                                                                                                                                                                                                                                                        
                  <appender name="FILE" class="...">
                      <file>${appName}/logs/${appName}.log</file>                                                                                                                                                                                          
                      ...                                                                                                                                                                                                                                  
                  </appender>                                                                                                                                                                                                                                           
                <springProperty> 

hooks into the Spring Environment after it's been populated, which is exactly when logback-spring.xml is processed. This works in both dev and production.

**Comment by jdaugherty** [comment]
I don't believe this is an issue - this is a misconfiguration with your log file per the above.  I'm going to close this ticket as such.

**Comment by gsartori** [comment]
That's great, thank you for the fix @jdaugherty
